File: frontend/zeal-edu/src/app/core/models/class-session.ts
export enum ClassSessionStatus {
  Scheduled = 'Scheduled',
  Completed = 'Completed',
  Cancelled = 'Cancelled',
}

export interface ClassSession {
  sessionId: string;
  batchId: string;
  sessionDate: string;
  startTime: string;
  endTime: string;
  topic: string | null;
  location: string | null;
  status: ClassSessionStatus;
  attendanceMarkedCount: number;
}

export interface CreateClassSessionPayload {
  sessionDate: string;
  startTime: string;
  endTime: string;
  topic: string | null;
  location: string | null;
}

export interface UpdateClassSessionPayload {
  sessionDate: string;
  startTime: string;
  endTime: string;
  topic: string | null;
  location: string | null;
  status: ClassSessionStatus;
}

File: backend/src/ZealEducation.Domain/Entities/ClassSession.cs
using ZealEducation.Domain.Common;
using ZealEducation.Domain.Enums;

namespace ZealEducation.Domain.Entities;

public class ClassSession : BaseAuditableEntity
{
    public Guid BatchId { get; set; }
    public DateOnly SessionDate { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public string? Topic { get; set; }
    public string? Location { get; set; }
    public ClassSessionStatus Status { get; set; } = ClassSessionStatus.Scheduled;

    public Batch Batch { get; set; } = default!;
    public ICollection<AttendanceRecord> AttendanceRecords { get; set; } = [];
}
